File: tmp/fix_competidores.js
const fs = require('fs');
const path = require('path');

const filePath = path.join('d:', 'repositorio_geral', 'app_gestao_contratos', 'src', 'components', 'CompetidoresSeccion.tsx');

if (!fs.existsSync(filePath)) {
  console.log(`File not found: ${filePath}`);
  process.exit(1);
}

let content = fs.readFileSync(filePath, 'utf-8');
const lines = content.split('\n');

if (lines.length > 26) {
    const line27 = lines[26];
    if (line27.includes("O BENCHMARK MÁXIMO") && line27.includes("position: 'absolute'")) {
        console.log("Found pattern on line 27");
        if (line27.includes("</div><div style")) {
            const parts = line27.split("</div><div style");
            lines[26] = parts[0] + "</div>";
            console.log("Updated line 27");
            
            // Remove lines 28 to 34 (indices 27 to 33 inclusive)
            for (let i = 27; i < 34; i++) {
                 if (i < lines.length) {
                      lines[i] = "";
                 }
            }
            
            fs.writeFileSync(filePath, lines.join('\n'), 'utf-8');
            console.log("Successfully updated file.");
        } else {
             console.log("Could not split line 27 accurately.");
        }
    } else {
         console.log("Pattern not found on line 27.");
    }
} else {
    console.log("File too short.");
}

File: tmp/fix_competidores.py
import os

file_path = r"d:\repositorio_geral\app_gestao_contratos\src\components\CompetidoresSeccion.tsx"

if not os.path.exists(file_path):
    print(f"File not found: {file_path}")
    exit(1)

with open(file_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Inspect line 27 (0-indexed 26)
if len(lines) > 26:
    line27 = lines[26]
    if "O BENCHMARK MÁXIMO" in line27 and "position: 'absolute'" in line27:
        print("Found pattern on line 27")
        # Split by the tag transition </div><div
        if "</div><div style" in line27:
            parts = line27.split("</div><div style")
            lines[26] = parts[0] + "</div>\n"
            print("Updated line 27")
            
            # Remove lines 28 to 34 (indices 27 to 33 inclusive)
            # Lines list: 
            # 27: <div style... (index 26) -> updated
            # 28: <iframe (index 27) -> remove
            # 34: </div> (index 33) -> remove
            for i in range(27, 34):
                if i < len(lines):
                    print(f"Removing line {i+1}: {lines[i].strip()}")
                    lines[i] = ""
                    
            with open(file_path, 'w', encoding='utf-8') as f:
                f.writelines(lines)
            print("Successfully updated file.")
        else:
            print("Could not split line 27 accurately.")
    else:
         print("Pattern 'O BENCHMARK MÁXIMO' or 'absolute' not on line 27.")
else:
    print("File too short.")
